src/pages/settings/Payments/PaymentsPage/BasePaymentsPage.js

```diff
@@ -235,18 +279,25 @@ class BasePaymentsPage extends React.Component {
 
     /**
      * Hide the default / delete modal
+     * @param {boolean} shouldClearSelectedData - Clear selected payment method data if true
      */
-    hideDefaultDeleteMenu() {
+    hideDefaultDeleteMenu(shouldClearSelectedData = true) {
         this.setState({shouldShowDefaultDeleteMenu: false});
         InteractionManager.runAfterInteractions(() => {
             this.setState({
                 showConfirmDeleteContent: false,
             });
+            if (shouldClearSelectedData) {
+                this.resetSelectedPaymentMethodData();
+            }
```

Comment:
I don't that's a good Idea logically because resetSelectedPaymentMethodData method is solely to reset the selected payment method data and not directly take any action on the modal closing status.
And resetSelectedPaymentMethodData is also called from multiple places which don't even have direct interaction with showConfirmDeleteContent.

And InteractionManager.runAfterInteractions inside another InteractionManager.runAfterInteractions isn't wrong as I have seen places in our app itself using the same. The second InteractionManager.runAfterInteractions won't have any effect here.

Reply:
One other thing to note here is we clear the data only if shouldClearSelectedData is true. We have flows which send it to false. So we would end up doing like this. Seems to be making the logic more complex to read.

```
if (shouldClearSelectedData) {
            this.resetSelectedPaymentMethodData();
        } else {
            InteractionManager.runAfterInteractions(() => {
                this.setState({
                    showConfirmDeleteContent: false,
                });
            });
        }
```


Anyways if we want this. I can make the change.